amundsen_application/static/js/components/SearchPage/index.tsx

```diff
@@ -135,7 +145,7 @@ export class SearchPage extends React.Component<SearchPageProps> {
         return newTab;
       case ResourceType.dashboard:
       default:
-        return this.props.selectedTab;
+        return null;
```

Comment:
What is the logic surrounding this needing to be `null`?

Reply:
This addresses the issue where a "New search does not reset tab." By default we were using the previous tab, but now we send an explicit null, meaning we did not select any tab for this search.

Reply:
`getSelectedTabByResourceType` should be a validator for the tab type. It's called before making any saga requests to indicate which tab, if any we should select. Sending `props.selectedTab` will carry over into the next state. If there is no tab in the URL, we shouldn't be sending on into the redux action.